src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -3458,6 +3458,17 @@ public ILocalSymbol GetDeclaredSymbol(CatchDeclarationSyntax catchDeclaration, C
                         break;
                     }
 
+                case BoundKind.ConditionalAccess:
+                    {
+                        var conditionalAccess = (BoundConditionalAccess)boundNode;
+                        if (conditionalAccess.AccessExpression is BoundExpression expression && expression.ExpressionSymbol is Symbol symbol)
+                        {
+                            symbols = ImmutableArray.Create(symbol);
+                            resultKind = boundNode.ResultKind;
+                        }
+                        break;
```

Comment:
i woudl be very wary about changing the compiler here.  is that really necessary?  tagging @dotnet/roslyn-compiler @jcouv

Reply:
@chris-schuetz is this necessary?  Given an expression like `a?[b]` you shoudl have a ConditionalAccessExpression whose `whenNotNull` component is an ElementBindingExpression. 

When you bind that ElementBindingExpression do you not get back the indexer symbol that is expected?

Reply:
Thanks for pointing me to the whenNotNull component. I undid these changes.